components/ca-export.js

```diff
@@ -0,0 +1,329 @@
+define([
+    './helpers/create-element.js',
+    './helpers/cancel-event.js',
+    './helpers/json-to-csv.js',
+    './helpers/component-support.js',
+    './helpers/post-via-iframe.js',
+    'document-register-element'
+], (createElement, cancelEvent, jsonToCSV, componentSupport, postViaIframe) => {
+    /**
+     * Class export
+     * @description A custom HTML element (Web Component) that can be created using
+     * document.createElement('ca-export') or included in a HTML page as an element.
+     */
+    class Export extends HTMLElement {
+        /**
+         * createdCallback - called when the component created, (not yet attached to the DOM).
+         * @returns {undefined} initalises the component.
+         */
+        createdCallback() {
+            this.showRecordCount = true;
+            // insert download button
+            this.button = createElement(
```

Comment:
I personally found the original to be less readable. So I thought on this instance I would run it via a linter + prettier.github.io/prettier/ with 140 width to avoid being bias. Do you prefer the original?